src/app/SharpRaven/Utilities/CircularBuffer.cs

```diff
@@ -42,4 +45,105 @@ public bool IsEmpty()
             return queue.IsEmpty;
         }
     }
+
+    internal class ConcurrentQueue<T> : ICollection, IEnumerable<T>
```

Comment:
If `ConcurrentQueue` is just an implementation detail of `CircularBuffer`, you can just put it in as a nested class. Otherwise, move it to its own file. Are you not missing an `#if (net35)` around the class declaration, by the way? We'd want to use `System.Collections.Concurrent.ConcurrentQueue` in .NET 4.0+ and not our custom implementation, no?